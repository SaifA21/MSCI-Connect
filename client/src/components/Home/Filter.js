
import React, { Component } from 'react';

import { withStyles } from '@material-ui/core/styles';
import CssBaseline from "@material-ui/core/CssBaseline";
import { MuiThemeProvider, createTheme } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import {Grid, AppBar, Box, Toolbar, Button, Select, MenuItem, FormControl, InputLabel, TextField, Radio, FormControlLabel, RadioGroup, FormLabel, FormHelperText, helperText, TableRow} from "@material-ui/core/";

import IconButton from '@material-ui/core/IconButton';

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';




const Filter = (props) => {
  

    return(
    <FormControl variant="filled" style={{minWidth: 300}}>
          <InputLabel id="sort">Filter by:</InputLabel>
          <Select
            labelId="sortBySelector"
            id="sortBySelector"
            //value={age}
            onChange={(event)=>{
                    
              props.mainPagefilter(event.target.value)
              console.log(event.target.value)
         
  
              
  
            }}
          >
            <MenuItem value=""><em>None</em></MenuItem>
            <MenuItem value={'MSCI 446'}>MSCI 446</MenuItem>
            <MenuItem value={'MSCI 431'}>MSCI 431</MenuItem>
            <MenuItem value={'MSCI 342'}>MSCI 342</MenuItem>
            <MenuItem value={'MSCI 334'}>MSCI 334</MenuItem>
            <MenuItem value={'MSCI 311'}>MSCI 311</MenuItem>
            <MenuItem value={'Co-op'}>Co-op</MenuItem>
            <MenuItem value={'General'}>General</MenuItem>
          </Select>
        </FormControl>
    )
  }

  export default (Filter)