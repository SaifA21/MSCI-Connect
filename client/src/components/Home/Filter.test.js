import renderer from 'react-test-renderer';
import Filter from './Filter.js';



//Unit test written by Suhayl Sayed

describe('Filter Component', () => {

    it('renders correctly', () => {
        const tree = renderer
          .create(<Filter></Filter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

})