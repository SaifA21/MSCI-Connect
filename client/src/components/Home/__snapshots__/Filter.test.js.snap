// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component renders correctly 1`] = `
<div
  className="MuiFormControl-root"
  style={
    Object {
      "minWidth": 300,
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
    id="sort"
  >
    Filter by:
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="sortBySelector sortBySelector"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
      id="sortBySelector"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
